cachi2/core/package_managers/gomod.py

```diff
@@ -83,7 +95,201 @@ class GoPackage(_GolangModel):
 
     import_path: str
     standard: bool = False
-    module: Optional[GoModule]
+    module: Optional[ParsedModule]
+
+
+class Module(NamedTuple):
+    """A Go module with relevant data for the SBOM generation.
+
+    name: the module's name, as written in go.mod
+    original_name: name prior to the replacement in case of a replaced module, equals to name otherwise
+    real_path: real path to locate the package on the Internet, which differs from name in case of a
+        local module if the main module's name does not match the repository's URL (e.g. forks)
+    version: the resolved module's version
+    main: if this is the main module in the repository subpath that is being processed
+    """
+
+    name: str
+    original_name: str
+    real_path: str
+    version: str
+    main: bool = False
+
+    @property
+    def purl(self) -> str:
+        """Get the purl for this module."""
+        return f"pkg:golang/{self.real_path}@{self.version}?type=module"
+
+    def to_component(self) -> Component:
+        """Create a SBOM component for this module."""
+        return Component(name=self.name, version=self.version, purl=self.purl)
+
+
+class Package(NamedTuple):
+    """A Go package with relevant data for the SBOM generation.
+
+    relative_path: name of the package relative to the parent module's name
+    module: parent module for this package
+    """
+
+    relative_path: Optional[str]
+    module: Module
+
+    @property
+    def name(self) -> str:
+        """Get the name for this package based on the parent module's name."""
+        if self.relative_path:
+            return f"{self.module.name}/{self.relative_path}"
+
+        return self.module.name
+
+    @property
+    def real_path(self) -> str:
+        """Get the real path to locate this package on the Internet."""
+        if self.relative_path:
+            return f"{self.module.real_path}/{self.relative_path}"
+
+        return self.module.real_path
+
+    @property
+    def purl(self) -> str:
+        """Get the purl for this package."""
+        return f"pkg:golang/{self.real_path}@{self.module.version}?type=package"
+
+    def to_component(self) -> Component:
+        """Create a SBOM component for this package."""
+        return Component(name=self.name, version=self.module.version, purl=self.purl)
+
+
+class StandardPackage(NamedTuple):
+    """A package from Go standard lib used in the SBOM generation.
+
+    Standard lib packages lack a parent module and, consequentially, a version.
+    """
+
+    name: str
+
+    @property
+    def purl(self) -> str:
+        """Get the purl for this package."""
+        return f"pkg:golang/{self.name}?type=package"
+
+    def to_component(self) -> Component:
+        """Create a SBOM component for this package."""
+        return Component(name=self.name, purl=self.purl)
```

Comment:
@brunoapimentel, @chmeliik; Is there nothing in the purl/SBOM to indicate that a package comes from the Go stdlib?

Reply:
I'd say the lack of a version is the main giveaway, but otherwise, nothing explicit

Reply:
^ that and the fact that the `name` in the purl doesn't start with a hostname (first element doesn't have a `.`)

Reply:
ACK